pkg/prometheus/promcfg.go

```diff
@@ -1804,3 +1819,16 @@ func (l *limitEnforcer) addLimitsToYAML(cfg yaml.MapSlice, k limitKey, limit uin
 
 	return append(cfg, yaml.MapItem{Key: k.prometheusField, Value: getLimit(limit, enforcedLimit)})
 }
+
+func (l *limitEnforcer) addBodySizeLimitsToYAML(cfg yaml.MapSlice, enforcedLimit string) yaml.MapSlice {
+	if enforcedLimit == "" {
+		return cfg
+	}
+
+	if l.prometheusVersion.LT(semver.MustParse("2.28.0")) {
+		level.Warn(l.logger).Log("msg", "body_size_limit is only available starting from prometheus 2.28.0",
+			"version", l.prometheusVersion)
+		return cfg
+	}
+	return append(cfg, yaml.MapItem{Key: "body_size_limit", Value: enforcedLimit})
+}
```

Comment:
@simonpasquier I'm not as familiar with this part of the code, but I wonder should this take on a similar pattern that we implemented in AlertManager config where we sanitized post config generation?

If it is applicable we could take care of it in a different PR

Reply:
It won't add this field if version doesn't have minimum required version. So we don't have to remove the fields if unsupported

Reply:
The Prometheus controller takes another approach: instead of duplicating the Prometheus config struct, it builds a `yaml.MapSlice` struct dynamically. Honestly I'm not sure which approach is the less terrible :)

Reply:
@simonpasquier  me neither, although I do see a lot of version checks scattered throughout the file so maybe consolidating them to a single function is preferable. 

@slashpai not a blocker here, was just curious if we should align the approach